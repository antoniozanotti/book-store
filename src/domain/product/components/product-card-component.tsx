import { UiTitleComponent } from "@/domain/ui/components/ui-title-component";
import { ProductWithPageAndBookPrisma } from "../types/product-prisma";
import { ProductLinkComponent } from "./product-link-component";
import { getAuthorsByProduct } from "../helpers/get-authors-by-product";
import { ProductListItemComponent } from "./product-list-item-component";
import { UiFigureComponent } from "@/ui-figure-component";

export function ProductCardComponent({
  product,
}: {
  product: ProductWithPageAndBookPrisma;
}) {
  return (
    <ProductListItemComponent>
      <ProductLinkComponent product={product}>
        <UiFigureComponent
          src={`product/${product.page.slug}`}
          width={187}
          originalWidth={640}
          height={269}
          alt={product.name}
          className="w-full sm:w-[187px] h-[249px] relative overflow-hidden"
          innerImgStyle="hover:-mt-[20px] w-full h-[269px] object-cover"
        />
      </ProductLinkComponent>
      <UiTitleComponent level="p" size="small" className="line-clamp-2 mt-2">
        <ProductLinkComponent product={product}>
          {product.name}
        </ProductLinkComponent>
      </UiTitleComponent>
      <address className="line-clamp-2">
        by {getAuthorsByProduct(product)}
      </address>
    </ProductListItemComponent>
  );
}
