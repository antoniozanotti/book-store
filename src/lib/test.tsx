import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
