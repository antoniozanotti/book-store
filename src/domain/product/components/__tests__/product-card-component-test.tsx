import { render, screen } from "@/lib/test";
import { ProductCardComponent } from "../product-card-component";
import { productWithPagEandBookMock } from "../../__tests__/product-with-page-and-book-mock";

describe("ProductCardComponent", () => {
  it("should render", async () => {
    const { container } = render(<ProductCardComponent product={productWithPagEandBookMock} />);
    expect(container).toMatchSnapshot();
  });
});