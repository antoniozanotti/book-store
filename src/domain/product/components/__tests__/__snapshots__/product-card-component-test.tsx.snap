// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <article
      class="p-5 w-full sm:w-[227px] h-[400px] text-sm text-pretty"
    >
      <a
        href="/product/the-four-agreements-a-practical-guide-to-personal-freedom"
        title="The Four Agreements: A Practical Guide to Personal Freedom by Don Miguel Ruiz"
      >
        <figure
          class="bg-green-300 w-full sm:w-[187px] h-[249px] relative overflow-hidden"
        >
          <img
            alt="The Four Agreements: A Practical Guide to Personal Freedom"
            class="transition-all hover:-mt-[20px] w-full h-[269px] object-cover"
            data-nimg="1"
            decoding="async"
            height="269"
            loading="lazy"
            sizes="640px"
            src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_5523,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0"
            srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_23,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_46,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_69,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_92,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_138,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_184,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_368,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_552,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_920,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_1078,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_1191,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1553,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1726,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_2761,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2946,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_5523,g_auto/f_auto/q_auto/v1/product/the-four-agreements-a-practical-guide-to-personal-freedom?_a=BAVARSDW0 3840w"
            style="color: transparent;"
            width="187"
          />
        </figure>
      </a>
      <p
        class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
      >
        <a
          href="/product/the-four-agreements-a-practical-guide-to-personal-freedom"
          title="The Four Agreements: A Practical Guide to Personal Freedom by Don Miguel Ruiz"
        >
          The Four Agreements: A Practical Guide to Personal Freedom
        </a>
      </p>
      <address
        class="line-clamp-2"
      >
        by 
        Don Miguel Ruiz
      </address>
    </article>
  </div>
</div>
`;
